### PR TITLE
feat: added `__get_property_by_name`-method and return `None` for `FeedbackDataset.*_by_name()`

### LINKED ISSUE #4101: [FEATURE] Align `FeedbackDataset.*_by_name` methods for better integration with conditional creation flows
**Is your feature request related to a problem? Please describe.**
The current behavior for the several `FeedbackDataset.*_by_name` method is to raise a `KeyError` when the entity is not found by name. Instead of that, we could just return a `None` value. This scenario could work when admin users want to control when a new entity must be created for an existing dataset:

#### What we need to do right now
```python
try:
   dataset.question_by_name("label_question")
except KeyError:
   dataset.add_question(rg.LabelQuestion(name="label_question",...)
```

**Describe the solution you'd like**

#### What I propose
```python
question = dataset.question_by_name("label_question")
if question is None:
   dataset.add_question(rg.LabelQuestion(name="label_question",...)
```
We're using questions as an example, but this can work with other entities.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by davidberenstein1957**
# Description

I agree with Paco that the implementation was a bit blocking. Instead we show a warning if not present and not block running code.

Closes #4101 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [X] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

N.A.

**Checklist**

- [X] I added relevant documentation
- [X] I followed the style guidelines of this project
- [X] I did a self-review of my code
- [X] I made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [X] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/client/feedback/dataset/base.py [hunk 1]
```diff
@@ -119,49 +119,43 @@ def allow_extra_metadata(self) -> bool:
         """Returns whether if adding extra metadata to the records of the dataset is allowed"""
         return self._allow_extra_metadata
 
+    def __get_property_by_name(self, item_name: str, iterable_items, item_type: str):
+        for item in iterable_items:
+            if item.name == item_name:
+                return item
+            if hasattr(item, "title"):
+                if item.title == item_name:
```
Comment (davidberenstein1957) [inline src/argilla/client/feedback/dataset/base.py]: @frascuchon, I added this in the assumption that names can't overlap with titles. Is that correct?

**Comment by davidberenstein1957** [comment]
I still need to add tests

**Comment by frascuchon** [comment]
I see this change when I update the similarity search feature branch from develop.  We must be conscience that changes on this PR are breaking changes and should be informed properly (in CHANGELOG and documentation). 

The reason behind the breaking changes is that we're changing the current behavior of the existing method and some code using previous versions could fail (if users are currently capturing the `KeyError`). 

In my opinion, in order to accept this change we should 
- 1. Include a [breaking] tag in the CHANGELOG
- 2. Update documentation sections referring to how to get fields/questions/...by name (and show how users should be from these changes.
- 3. Prepare a section for the Release notes about his breaking change (The `field_by_name` method has changed, and now should handle by using `None` bla blab lab...)

Please @davidberenstein1957, take care of this to not create unexpected confusion with upgrading the version

cc @nataliaElv

**Comment by davidberenstein1957** [comment]
Hi Paco,Thanks for the feedback. I didn't look at it this way. I will make sure to change this. Also, it might be an idea to slowly build the release highlight in the changelog during PRs and then just copy/paste when applicable?On 7 Nov 2023, at 00:37, Francisco Aranda ***@***.***> wrote:﻿
I see this change when I update the similarity search feature branch from develop.  We must be conscience that changes on this PR are breaking changes and should be informed properly (in CHANGELOG and documentation).
The reason behind the breaking changes is that we're changing the current behavior of the existing method and some code using previous versions could fail (if users are currently capturing the KeyError).
In my opinion, in order to accept this change we should



Include a [breaking] tag in the CHANGELOG




Update documentation sections referring to how to get fields/questions/...by name (and show how users should be from these changes.




Prepare a section for the Release notes about his breaking change (The field_by_name method has changed, and now should handle by using None bla blab lab...)



Please @davidberenstein1957, take care of this to not create unexpected confusion with upgrading the version
cc @nataliaElv

**Comment by nataliaElv** [comment]
Has this been properly documented? Docs should be reviewed so that there is no code that becomes irrelevant.

**Pull request merged**